File: code/1047/1047.rb
input = gets.split(' ')

initial_hour = input[0].to_i
initial_minute = input[1].to_i
final_hour = input[2].to_i
final_minute = input[3].to_i

initial_time = initial_hour * 60 + initial_minute
final_time = final_hour * 60 + final_minute

duration_time = initial_time < final_time ? final_time - initial_time : (60 * 24) - initial_time + final_time

duration_hour = duration_time / 60
duration_minute = duration_time % 60

puts "O JOGO DUROU #{duration_hour} HORA(S) E #{duration_minute} MINUTO(S)"

File: code/1047/1047.c
#include <stdio.h>

int main() {

  int initialHour, initialMinute, finalHour, finalMinute, initialTime, finalTime, durationTime, durationHour, durationMinute;

  scanf("%d %d %d %d", &initialHour, &initialMinute, &finalHour, &finalMinute);

  initialTime = initialHour * 60 + initialMinute;
  finalTime = finalHour * 60 + finalMinute;

  if (initialTime < finalTime) {
    durationTime = finalTime - initialTime;
  } else {
    durationTime = (60 * 24) - initialTime + finalTime;
  }

  durationHour = durationTime / 60;
  durationMinute = durationTime % 60;

  printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", durationHour, durationMinute);

  return 0;
}
